### PR TITLE
[snap]: Added import account feature

### PR DESCRIPTION
**Pull request by AnthonyLaw**
## What was the issue?
- The user is not allowed to import private in the wallet.

## What's the fix?
- Added import account logic, allowing the user to provide an account name and private key for import into snap state.
    - Once the private key is imported, MetaMask's confirmation box will pop up, and the user needs to approve it.
    - If the private key is already imported, MetaMask's alert box will pop up.
- Refactor: removed `components/AccountCreateModalBox` and replaced it with `AccountCreationFormModalBox` which can be shared between `Create/Import` forms.
- Installed Symbol-sdk to frontend project.
- Added an `import` label in accounts list.

## Screenshot

Screenshot 2024-06-25 at 5 20 02 PM

Screenshot 2024-06-25 at 5 19 53 PM

image

image

### CODE REVIEW

**Review by Wayonb — APPROVED** [review]
👍🏾

File: snap/backend/src/utils/accountUtils.js [hunk 1]
```diff
@@ -70,34 +112,64 @@ const accountUtils = {
...
 			throw new Error(`Failed to create account: ${error.message}`);
 		}
+	},
+	async importAccount({ state, requestParams }) {
+		try {
+			const { privateKey, accountLabel } = requestParams;
+			const { accounts, network } = state;
+
+			const keyPair = new SymbolFacade.KeyPair(new PrivateKey(privateKey));
+			const existingAccount = Object.values(accounts).find(account => account.privateKey === keyPair.privateKey.toString());
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply snap/backend/src/utils/accountUtils.js:130]: I made update.

**Review by Wayonb — CHANGES_REQUESTED** [review]

File: snap/backend/src/utils/accountUtils.js [hunk 2]
```diff
@@ -61,6 +65,53 @@ const accountUtils = {
...
+	 * @param {object} state - The snap state object.
+	 * @param {SymbolFacade} facade - Symbol facade instance.
+	 * @param {SymbolFacade.KeyPair} keyPair - The key pair object.
+	 * @param {string} accountLabel - The account label.
+	 * @param {'metamask' | 'import'} type - The snap account type.
+	 * @param {number} addressIndex - The address index.
+	 * @returns {Account} - The account object.
+	 */
+	async createAccountAndUpdateState({
+		state, facade, keyPair, accountLabel, type, addressIndex = null
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply snap/backend/src/utils/accountUtils.js:79]: oh yeah, thanks for remind 👍🏼

File: snap/backend/src/utils/accountUtils.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+	copyable, heading, panel, text
+} from '@metamask/snaps-sdk';
 import { PrivateKey } from 'symbol-sdk';
 import { SymbolFacade } from 'symbol-sdk/symbol';
 import { v4 as uuidv4 } from 'uuid';
 
 const AccountType = {
-	METAMASK: 'metamask'
+	METAMASK: 'metamask',
+	IMPORTED: 'import'
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply snap/backend/src/utils/accountUtils.js:12]: import: The account is generated from the private key given by the user.
metamask: The account is generated from bip44 and derives the private key from metamask.

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**